ext/src/ruby_api/store.rs

```diff
@@ -170,6 +172,64 @@ impl Store {
         Ok(())
     }
 
+    /// @yard
+    /// Sets limits on the {Store}.
+    /// @param memory_size [Integer]
+    ///   The maximum number of bytes a linear memory can grow to.
+    /// @param table_elements [Integer]
+    ///   The maximum number of elements in a table.
+    /// @param instances [Integer]
+    ///   The maximum number of instances that can be created for a Store.
+    /// @param tables [Integer]
+    ///   The maximum number of tables that can be created for a Store.
+    /// @param memories [Integer]
+    ///   The maximum number of linear memories that can be created for a Store.
+    /// @raise [Error] if arguments can not be converted.
+    pub fn set_limits(&self, args: &[Value]) -> Result<(), Error> {
```

Comment:
Even though we had discussed using `wasmtime-py`'s approach here, upon a deeper look I do wonder if it'll be better if instead of using the `set_limiter` method, we allow the limits to be optionally passed when creating the store, (in `Store#new`) similar to how the Wasi context is passed in. This will make this more uniform with the current API IMHO. I _think_ that `wasmtime-py` convention is to favor the `set_*` approach for most of the APIs (e.g. it exposes a `set_wasi` method for example).

Reply:
I've updated this to set the limits in the constructor - I'm not sure whether there is a simpler way to call the `limiter` method from within the constructor and whether `StoreData.store_limits` is necessary any more.